app/learning/p_hosts.py

```diff
@@ -0,0 +1,17 @@
+import re
+
+from app.objects.secondclass.c_fact import Fact
+
+
+class Parser:
+
+    def __init__(self):
+        self.trait = 'multiple'  # remote.host.name, remote.host.fqdn
```

Comment:
my gut feeling is that every parser should be tied to 1 (generic) trait and something higher up the chain should adjust it to make it more specific if needed

Reply:
so in that case would the one specific trait be the end of the little triple created by the current traits?  eg (`*.*.data_type` where 'data_type' is the targeted element of the parser and the prior elements give context and location to the data/fact source)

Reply:
so in the cases above it would actually be two parsers, one for 'name' and one for 'fqdn', and that we would have a way to programatically set the populated context elements, either in the parser itself based on specific parsing rules, or based on the ability ran and what domain its command output would applicable to?